### ISSUE TITLE
Uso de make_dirty()

### ISSUE THREAD

**Issue by drcid98** [DiegoEmilio01/IIC3413#9]
Hola! Tengo una duda sobre el uso del make_dirty(). Resulta que en el enunciado se habla de que hay que manejar bien su uso, pero se dice solo en la sección del método vacuum:

![2024-04-09_22-00](https://github.com/DiegoEmilio01/IIC3413/assets/39897206/c301246b-7c57-46bd-946c-a53530722d0c)

Esto a pesar de ahí mismo se dice que make_dirty() debe ser usado siempre que una página es modificada. Dado esto, se debería también usar make_dirty() cuando la inserción de una tupla es exitosa y cuando se elimina una tupla, cierto?

Gracias!

**Comment by DiegoEmilio01** [comment]
Sí, deben utilizar ese método cada vez que deseen que un cambio sea persistente en la BBDD. Las 3 funciones que se pide implementar deben ser persistentes. Pueden usar el _make_dirty()_ de forma lazy o de forma inteligente sólo en los casos necesarios (esto no es evaluado, solo que siempre sea persistente).

**Comment by drcid98** [comment]
Súper, gracias! una duda de elo que dijiste, a qué te refieres con la forma lazy o inteligente?

**Comment by DiegoEmilio01** [comment]
Por ejemplo, en el método _delete_record_ es necesario marcar _dirty_ siempre? Podrías usar _dir_deleted_ para revisar si ya está borrado. De esa forma te ahorras una operación en disco muy costosa y tu BBDD será más rápida en la práctica. Si bien esto no se evalúa, es bueno que empiecen a obtener esa intuición.
